docs/content/tutorials/create-rest-api.md

```diff
@@ -0,0 +1,299 @@
+
+We also need to clean up the template. Remove the `error_handler (text "Api 404")` line inside Router.fs. Remove `open FSharp.Control.Tasks.ContextInsensitive` from BookRepository.fs and BookController.fs file inside the Books folder. Add `open Giraffe` to BookController.fs. Finally, replace `open Giraffe.GiraffeViewEngine` with `open Giraffe.ViewEngine` inside BooksViews.fs.
+
+## Configure Endpoint Routing
+
+We will be using Endpoint Routing for the application. See [here](https://saturnframework.org/explanations/endpoint-routing.html) for more information. To do this. Add `open Saturn.Endpoint` to the Router.fs and BooksController.fs files. Then inside Program.fs, replace `use_router Router.appRouter` with `use_endpoint_router Router.appRouter`.
+
+## API Router
+
+We need to create router for our API endpoints. Create the following `apiRouter` at the bottom inside BooksController.fs to handle the standard GET, POST, PUT, DELETE.
+
+```fsharp
+let apiRouter =
+    router {
+        get "/" getAction
+        getf "/%s" (fun id -> getByIdAction id)
+        post "/" (bindJson<Book> (fun book -> postAction book))
+        putf "/%s" (fun id -> (bindJson<Book> (fun book -> putAction id book)))
+        deletef "/%s" (fun id -> deleteApiAction id)
+    }
+```
+
+With the setup done earlier, this will set the base of our Books API endpoints at `"http://localhost:8085/api/books/"`.
+
+We have not create the action functions yet. This router will automatically pass `HttpFunc` and `HttpContext` to our functions. All our action functions will have those as our parameter.
+
+For GET, we need no id to get all book so the route will just the current location. Because of `forward "/books" Books.Controller.apiRouter` inside our Router.fs file. The path for our GET request is `"http://localhost:8085/api/books/"`
+
+To get a specific book by id, we will need to get our id from the URL. For that, we will use `getf` and the format string `"/%s"` to read the string from the URL and pass it on to our function. You can also write this as `getf "/%s" getByIdAction`. Refer to the table below to modify the format string to your need (e.g. use `"/%i"` if your id is an int). So to get the book with the id of 2, create a GET request to `"http://localhost:8085/api/books/2"`
+
+| Format String | Type |
+| ----------- | ---- |
+| `%b` | `bool` |
+| `%c` | `char` |
+| `%s` | `string` |
+| `%i` | `int` |
+| `%d` | `int64` |
+| `%f` | `float`/`double` |
+| `%O` | `Guid` |
+
+For POST, we need to parse the object passed in with the request. Generally, this is a json object so `bindJson` is used. To test, send a POST request to `"http://localhost:8085/api/books/"` with the object inside the body as JSON.
+
+For PUT, we need both the id and the object so we will have both `putf` and `bindJson` to get the values to pass to `putAction`.
+
+The id is needed for DELETE so we use `deletef`.
+
+## GET

```

Comment:
I think maybe this could be merged with the GET router explainer above? Same with POST, PUT, and DELETE?